examples/webgpu_tsl_procedural_terrain.html

```diff
@@ -0,0 +1,357 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js webgpu - procedural terrain</title>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js webgpu</a> - procedural terrain
+			<br>
+			Based on <a href="https://threejs-journey.com/lessons/procedural-terrain-shader" target="_blank" rel="noopener">Three.js Journey</a> lessons
+		</div>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.webgpu.js",
+					"three/tsl": "../build/three.webgpu.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+			import { mx_noise_float, color, cross, dot, float, modelNormalMatrix, positionLocal, sign, step, tslFn, uniform, varying, vec2, vec3, loop } from 'three/tsl';
+
+			import { GUI } from 'three/addons/libs/lil-gui.module.min.js';
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import { RGBELoader } from 'three/addons/loaders/RGBELoader.js';
+
+			let camera, scene, renderer, controls, drag;
+
+			init();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 35, window.innerWidth / window.innerHeight, 0.1, 100 );
+				camera.position.set( - 10, 8, - 2.2 );
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x201919 );
+
+				const gui = new GUI();
+
+				// environment
+
+				const rgbeLoader = new RGBELoader();
+				rgbeLoader.load( './textures/equirectangular/pedestrian_overpass_1k.hdr', ( environmentMap ) => {
+
+					environmentMap.mapping = THREE.EquirectangularReflectionMapping;
+
+					scene.background = environmentMap;
+					scene.backgroundBlurriness = 0.5;
+					scene.environment = environmentMap;
+			
+				} );
+
+				// lights
+			
+				const directionalLight = new THREE.DirectionalLight( '#ffffff', 2 );
+				directionalLight.position.set( 6.25, 3, 4 );
+				directionalLight.castShadow = true;
+				directionalLight.shadow.mapSize.set( 1024, 1024 );
+				directionalLight.shadow.camera.near = 0.1;
+				directionalLight.shadow.camera.far = 30;
+				directionalLight.shadow.camera.top = 8;
+				directionalLight.shadow.camera.right = 8;
+				directionalLight.shadow.camera.bottom = - 8;
+				directionalLight.shadow.camera.left = - 8;
+				directionalLight.shadow.normalBias = 0.05;
+				directionalLight.shadow.bias = 0;
+				scene.add( directionalLight );
+
+				// terrain
+
+				const material = new THREE.MeshStandardNodeMaterial( {
+					metalness: 0,
+					roughness: 0.5,
```

Comment:
Maybe set `side: THREE.DoubleSide` -- or constrain polar angle of `OrbitControls`? Your choice.

Reply:
Good idea.
I'd rather go for the polar angle since the experience is already quite heavy in terms of performance.

Reply:
OK, sounds good. Thank you for this beautiful example!